state/filesystem.go

```diff
@@ -769,6 +780,83 @@ func removeFilesystemOps(st *State, filesystem Filesystem, assert interface{}) (
 	return ops, nil
 }
 
+// ImportFilesystem imports an existing, already-provisioned
+// filesystem into the model. The model will start out with
+// the status "detached". The filesystem and associated backing
+// volume (if any) will be associated with the given storage
+// name, with the allocated storage tag being returned.
+func (st *State) ImportFilesystem(
+	info FilesystemInfo,
+	backingVolume *VolumeInfo,
+	storageName string,
+) (_ names.StorageTag, err error) {
+	defer errors.DeferredAnnotatef(&err, "cannot import filesystem")
+	if info.Pool == "" {
+		return names.StorageTag{}, errors.NotValidf("empty pool name")
+	}
+	if backingVolume == nil {
```

Comment:
I've separated the validation bits into a separate function, called validateAddExistingFilesystem. It's not FilesystemInfo.Validate, because it requires access to *State to use the poolmanager/provider to check support.